frontend/js/app.js

```diff
@@ -0,0 +1,54 @@
+// Home screen component
+const HomeComponent = {
+  render: () => homeScreen()
+}
+
+// Edit Schema screen component
+const SchemaComponent = {
+  render: () => schemaReport()
+}
+
+// Instructions Component
+const InstructionsComponent = {
+  render: () => renderInstructionsHtml()
+}
+
+// Error component (for any unrecognized path)
+const ErrorComponent = {
+  render: () => {
+    return `
+      <section>
+        <h1>Error</h1>
+      </section>
+    `;
+  }
+}
+
+// Pre defined routes 
+const routes = [
+  { path: '/', component: HomeComponent, },
+  { path: '/schema-report-connect-to-db', component: SchemaComponent, },
```

Comment:
These cannot be strings. These need to be variables because you are checking against them later. You can use an enum or const struct(for js) to represent these. For example 
`const paths = {
  connectToDb: "/schema-report-connect-to-db",
  ...
}`
The string "schema-report-connect-to-db" can only appear once in the entire codebase. This is to allow for changes to be made easily to the URL in the future. For example, let's say we don't want to use the term "session" anymore we just have to change this enum and that's it.

Reply:
`const paths` is now declared in app.js file which contains variable names corresponding to each possible path. And in all other places strings are replaced with these variable names